include/bamit/IntervalNode.hpp

```diff
@@ -199,80 +199,77 @@ void construct_tree(std::unique_ptr<IntervalNode> & node, std::vector<Record> co
 {
     if (!root)
     {
         return;
     }
 
     Position cur_median = root->get_median();
-    // If the current median is overlapping the read, add all records from this node and search the left and right.
-    input.seek(root->get_file_offset());
-    std::vector<Record> current_reads{};
-
-    for (auto const & r : input | properly_mapped)
-    {
-        if (std::make_tuple(r.reference_id(), r.reference_position()) > cur_median)
-        {
-            break;
-        }
-        current_reads.emplace_back(std::make_tuple(r.reference_id().value(), r.reference_position().value()),
-                                   std::make_tuple(r.reference_id().value(), r.reference_position().value() +
-                                                                             get_length(r.cigar_sequence())),
-                                   r.file_offset());
-    }
+    // If the current median is overlapping the read, take the current offset as result and search the left further.
     if (cur_median >= start && cur_median <= end)
     {
-        results.insert(std::end(results), std::begin(current_reads), std::end(current_reads));
-        overlap(input, root->get_left_node(), start, end, results);
-        overlap(input, root->get_right_node(), start, end, results);
+        offstream_pos = root->get_file_offset();
+        overlap(root->get_left_node(), start, end, offstream_pos);
     }
-    // If current median is to the right of the overlap, sort reads in ascending order and add all reads which
-    // start before the overlap ends.
+    // If current median is to the right of the overlap, go to the left tree.
     else if (end < cur_median)
     {
-        std::sort(current_reads.begin(), current_reads.end(), RecordComparatorStart());
-        for (auto const & r : current_reads)
-        {
-            if (r.start <= end)
-            {
-                results.push_back(r);
-            }
-            else
-            {
-                break;
-            }
-        }
-        overlap(input, root->get_left_node(), start, end, results);
+        offstream_pos = root->get_file_offset();
+        overlap(root->get_left_node(), start, end, offstream_pos);
     }
+    // If current median is to the left of the overlap, go to the right tree.
     else if (start > cur_median)
     {
-        std::sort(current_reads.begin(), current_reads.end(), RecordComparatorEnd());
-        for (auto const & r : current_reads)
+        offstream_pos = root->get_file_offset();
+        overlap(root->get_right_node(), start, end, offstream_pos);
+    }
+}
+
+/*!
+   \brief Find the records which overlap a given start and end position.
+   \param input The sam file input of type bamit::sam_file_input_type.
+   \param root The root of the tree to search in.
+   \param start The start position of the search.
+   \param end The end position of the search.
+   \param outname The output filename.
+*/
+void get_records(sam_file_input_type & input,
```

Comment:
I think we should rename this function to `get_overlap_records`, since this is what a user would expect the function to be named.

The recursive function could be something like `get_overlap_file_offset`? Then the function names are much more descriptive than `get_records` and `overlap`.